drivers/modem/ublox-sara-r4.c

```diff
@@ -377,6 +390,9 @@ static int send_data(struct modem_socket *sock,
 	/* Loop through packet data and send */
 	while (frag) {
 		if (frag->len > 0) {
+#if defined(CONFIG_MODEM_UBLOX_SARA_U2)
+			mdm_receiver_send(&ictx.mdm_ctx, frag->data, frag->len);
```

Comment:
After seeing this, I realize that the existing logic is completely broken for the HEX sending on SARA-R4.  I'm using a buffer that is only sized correctly for the first send (AT+USO**=) and not the HEX encoded portion.   We should be able to make this work for both variants.

Reply:
Ok, maybe it isn't "completely broken" as the data being sent is only 2 bytes at a time.  But this needs another pass.  I'll see if I can come up with something better.

Reply:
I removed that entire section for the U2, as it works just fine using as bit-stream rather than a Hex conversion.

Reply:
Yep, I confirmed that HEX encoding isn't needed -- I misunderstood the spec when reading the sendto/recvfrom AT commands.  I'll put up a PR to remove the HEX encode / decode functionality.